src/gui/PageLoader.qml

```diff
@@ -61,6 +71,22 @@ HLoader {
         return true
     }
 
+    function moveThroughVisitHistory(relativeMovement=1) {
```

Comment:
We can shorten the name to `moveThroughHistory`. I think this can replace the `showPrevious` function too.

Reply:
`moveThroughHistory` and `showPrevious` have different checks, one acknowledges `historyPosition` while the other ignores it (because it will be reset). I could merge these two functions and add an argument for picking between one and the other behavior, but I think that would make the code less neatly organized. What do you think?

Reply:
I'm bothered by how the two functions end with the same 4 lines, but you can keep it that way for now I suppose

Reply:
I grouped that part into a new function that gets used in 2 places. I couldn't think of anything better.